Update gradle build configuration to support multi-project build

bhavikkumar opened the pull request:
This is to support gradle multi projects so that we can build client libraries which we can then push to private or public repositories for use in other services/projects.

bhavikkumar committed:
Update dependencies

bhavikkumar committed:
Update source code and target compatibility to Java 11

bhavikkumar committed:
Update dockerfile to use latest point release of Java 11

bhavikkumar committed:
Update NewRelic agent JAR to latest version

bhavikkumar committed:
Initial restructure for multiproject gradle build

bhavikkumar committed:
Move common testing dependencies to the root

bhavikkumar committed:
Move common settings to root build.gradle file

bhavikkumar committed:
Add client subproject and refactor gradle build

Removed superfluous gradle settings

bhavikkumar committed:
Update Travis-CI to use JDK11 but also build against early access

alphafoobar commented on build.gradle:
```diff
@@ -1,116 +1,70 @@
+
+    group = 'com.bnc'
+    version = '0.0.1-SNAPSHOT'
+    sourceCompatibility = 1.11
```
Perhaps use https://docs.gradle.org/current/javadoc/org/gradle/api/JavaVersion.html#VERSION_11

bhavikkumar replied:
Modified to use the object now

bhavikkumar committed:
Use JavaVersion when configuration source and target compatibility

bhavikkumar committed:
Update to using Java 11 options in Dockerfile

Reduced default container memory as Java 11 allows the java process to
consume 80% of the memory allocated instead of 50%

bhavikkumar committed:
Fix vulnerabilities identified by SonarQube

bhavikkumar committed:
Fix checkstyle issue

bhavikkumar committed:
Stop logging of query string as it is unsafe, use spring actuator

The information is available safely on /actuator/httptrace

bhavikkumar committed:
Run sonarqube in after_success phase

bhavikkumar committed:
Move sonarqube gradle task to test stage

bhavikkumar committed:
Update to gradle wrapper to 5.3.1
